osu.Game/Database/RealmLive.cs

```diff
@@ -102,10 +102,11 @@ public T Value
                 if (originalDataValid)
                     return data;
 
-                T retrieved;
+                if (!isCorrectThread)
+                    throw new InvalidOperationException($"Can't use {nameof(Value)} unless on the same thread the original data was fetched from.");
 
-                using (var realm = Realm.GetInstance(data.Realm.Config))
-                    retrieved = realm.Find<T>(ID);
+                var realm = Realm.GetInstance(data.Realm.Config);
+                var retrieved = realm.Find<T>(ID);
 
                 if (!retrieved.IsValid)
                     throw new InvalidOperationException("Attempted to access value without an open context");
```

Comment:
This check will never fail now. The whole point is that it was guaranteed to occur after the using block disposes of the Realm.

Reply:
Yep, correct. Can be commented for now if preferred.

Reply:
Definitely agree with leaving some sort of info that the exception branch is unreachable.